front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -48,123 +58,448 @@ interface DocumentOrTableUploadOrEditModalProps {
   viewType: ContentNodesViewType;
 }
 
-interface TableOrDocument {
+export function DocumentOrTableUploadOrEditModal(
+  props: DocumentOrTableUploadOrEditModalProps
+) {
+  const isTable = props.viewType === "tables";
+  const initialId = props.contentNode?.internalId;
+
+  const { table, isTableError, isTableLoading } = useTable({
+    owner: props.owner,
+    dataSourceView: props.dataSourceView,
+    tableId: isTable ? initialId ?? null : null,
+    disabled: !isTable,
+  });
+
+  const { document, isDocumentError, isDocumentLoading } =
+    useDataSourceViewDocument({
+      owner: props.owner,
+      dataSourceView: props.dataSourceView,
+      documentId: !isTable ? initialId ?? null : null,
+      disabled: isTable,
+    });
```

Comment:
Why aren't theses hooks not directly in the respective components ?

Reply:
It looks like you just need to pass dataSourceView, initialId and owner to the 2 components.

Reply:
Very fair 👍🏼